Sources/StreamChat/Utils/Database/NSManagedObject+Extensions.swift

```diff
@@ -33,20 +33,47 @@ extension NSFetchRequestGettable where Self: NSManagedObject {
 
     static func fetchRequest(keyPath: String, equalTo value: String) -> NSFetchRequest<Self> {
         let request = NSFetchRequest<Self>(entityName: entityName)
+        Self.applyPrefetchingState(to: request)
         request.predicate = NSPredicate(format: "%K == %@", keyPath, value)
         return request
     }
 }
 
 extension NSManagedObject: NSFetchRequestGettable {}
 
+// MARK: - Fetch Request Prefetching
+
+protocol NSFetchRequestPrefetching {}
+
+extension NSFetchRequestPrefetching where Self: NSManagedObject {
+    /// Turns off Core Data object faulting and sets prefetched relationship keypaths.
+    ///
+    /// Note: Reduces additional Core Data fetches when most of the data is accessed from fetched objects.
+    static func applyPrefetchingState(to request: NSFetchRequest<Self>) {
+        guard StreamRuntimeCheck._isDatabasePrefetchingEnabled else { return }
+        request.returnsObjectsAsFaults = false
+        request.relationshipKeyPathsForPrefetching = Self.prefetchedRelationshipKeyPaths()
+    }
+}
```

Comment:
This method is not in the `NSFetchRequest` extension because extending generic Obj-C classes is complicated.

It would be nicer to write `request.applyPrefetchingState()`, but did not find any good way for it which enables each subclass to define its relationship key paths. Moreover, load methods in this file make it even more complicated.